Update property-reference (#2161)
Kong/docs.konghq.com: app/enterprise/1.5.x/property-reference.md

logs is adjusted by the `log_level` directive.

#### status_access_log

**Default:** `off`

Path for Status API request access logs. The default value of `off` implies

> that loggin for this API is disabled by default.

Reviewer: ```suggestion
that login for this API is disabled by default.
```

If this value is a relative path, it will be placed under the `prefix`
location.

#### status_error_log

**Default:** `logs/status_error.log`

Path for Status API request error logs. The granularity of these logs is
adjusted by the `log_level` property.

### plugins

**Default:** `bundled`

**Description:**

> Comma-separated list of plugins on a node should load. By default, only plugins

Reviewer: ```suggestion
Comma-separated list of plugins on a node that should load. By default, only plugins
```

bundled in official distributions are loaded via the `bundled` keyword.

Loading a plugin does not enable it by default, but only instructs Kong to load
its source code, and allows to configure the plugin via the various related
Admin API endpoints.

> The specified name(s) will be substituted as such in the Lua namespace:

Reviewer: ```suggestion
The specified names will be substituted as such in the Lua namespace:
```

`kong.plugins.{name}.*`.

When the `off` keyword is specified as the only value, no plugins will be
loaded.

`bundled` and plugin names can be mixed together, as the following examples
suggest:

- `plugins = bundled,custom-auth,custom-log` will include the bundled plugins
  plus two custom ones
- `plugins = custom-auth,custom-log` will *only* include the `custom-auth` and
  `custom-log` plugins.
- `plugins = off` will not include any plugins

> **Note:** Kong will not start if some plugins were previously configured (i.e.

Reviewer: ```suggestion
**Note:** Kong will not start if some plugins were previously configured (i.e.,
```

have rows in the database) and are not specified in this list. Before disabling
a plugin, ensure all instances of it are removed before restarting Kong.

**Note:** Limiting the amount of available plugins can improve P99 latency when

> experiencing LRU churning in the database cache (i.e. when the configured

Reviewer: ```suggestion
experiencing LRU churning in the database cache (i.e., when the configured
```

Some suffixes can be specified for each pair:
- `ssl` will require that all connections made
   through a particular address/port be made with TLS
   enabled.
- `http2` will allow for clients to open HTTP/2
   connections to Kong's proxy server.
- `proxy_protocol` will enable usage of the
   PROXY protocol for a given address/port.

> - `transparent` will cause kong to listen to, and

Reviewer: ```suggestion
- `transparent` will cause Kong to listen to, and
```

   respond from, any and all IP addresses and ports
   you configure in iptables.
- `deferred` instructs to use a deferred accept on

>    Linux (the TCP_DEFER_ACCEPT socket option).

Reviewer: ```suggestion
   Linux (the `TCP_DEFER_ACCEPT` socket option).
```

- `bind` instructs to make a separate bind() call
   for a given address:port pair.
- `reuseport` instructs to create an individual

>    listening socket for each worker process

Reviewer: ```suggestion
   listening socket for each worker process,
```

>    allowing a kernel to distribute incoming

Reviewer: ```suggestion
   which allows a kernel to distribute incoming
```

>    connections between worker processes

Reviewer: ```suggestion
   connections between worker processes.
```

### headers

**Default:** `server_tokens, latency_tokens`

**Description**
Comma-separated list of headers Kong should inject in client responses.

Accepted values are:

> - `Server`: Injects `Server: kong/x.y.z` on Kong-produced response (e.g. Admin

Reviewer: ```suggestion
- `Server`: Injects `Server: kong/x.y.z` on Kong-produced response (e.g., Admin
```

  API, rejected requests from auth plugin).
- `Via`: Injects `Via: kong/x.y.z` for successfully proxied requests.
- `X-Kong-Proxy-Latency`: Time taken (in milliseconds) by Kong to process a
  request and run all plugins before proxying the request upstream.

> - `X-Kong-Response-Latency`: time taken (in millisecond) by Kong to produce a

Reviewer: ```suggestion
- `X-Kong-Response-Latency`: Time taken (in milliseconds) by Kong to produce a
```

suggest removing parens as well

>   response in case of e.g. plugin short-circuiting the request, or in in case of

Reviewer: ```suggestion
  response in case of a plugin short-circuiting the request, or in in case of
```

  an error.

> - `X-Kong-Upstream-Latency`: Time taken (in milliseconds) by the upstream

Reviewer: ```suggestion
- `X-Kong-Upstream-Latency`: Time taken (in milliseconds) by the Upstream
```

  service to send response headers.
- `X-Kong-Admin-Latency`: Time taken (in milliseconds) by Kong to process an
  Admin API request.

> - `X-Kong-Upstream-Status`: The HTTP status code returned by the upstream

Reviewer: ```suggestion
- `X-Kong-Upstream-Status`: The HTTP status code returned by the Upstream
```

>   service. This is particularly useful for clients to distinguish upstream

Reviewer: ```suggestion
  service. This is particularly useful for clients to distinguish Upstream
```

- **Cassandra**: 2.2 and above.

When not using a database, Kong is said to be in "DB-less mode": it will keep
its entities in memory, and each node needs to have this data entered via a
declarative configuration file, which can be specified through the
`declarative_config` property, or via the Admin API using the `/config`
endpoint.

#### database

Determines which of PostgreSQL or Cassandra this node will use as its
datastore.

Accepted values are `postgres`, `cassandra`, and `off`.

Default: `postgres`

---

#### Postgres settings

name   | description  | default
-------|--------------|----------
**pg_host** | Host of the Postgres server. | `127.0.0.1`
**pg_port** | Port of the Postgres server. | `5432`
**pg_timeout** | Defines the timeout (in ms), for connecting, reading and writing. | `5000`
**pg_user** | Postgres user. | `kong`
**pg_password** | Postgres user's password. | none
**pg_database** | The database name to connect to. | `kong`
**pg_schema** | The database schema to use. If unspecified, Kong will respect the `search_path` value of your PostgreSQL instance. | none
**pg_ssl** | Toggles client-server TLS connections between Kong and PostgreSQL. | `off`

> **pg_ssl_required** | When `pg_ssl` is on this determines if TLS must be used between Kong and PostgreSQL. | `off`

Reviewer: ```suggestion
**pg_ssl_required** | When `pg_ssl` is on, this setting determines if TLS must be used between Kong and PostgreSQL. | `off`
```

**pg_semaphore_timeout** | Defines the timeout (in ms) after which PostgreSQL query semaphore resource acquisition attempts will fail. Such failures will generally result in the associated proxy or Admin API request failing with an HTTP 500 status code. Detailed discussion of this behavior is available in the online documentation. | `60000`

#### Cassandra settings

name   | description  | default
-------|--------------|----------
**cassandra_contact_points** | A comma-separated list of contact points to your cluster. You may specify IP addresses or hostnames. Note that the port component of SRV records will be ignored in favor of `cassandra_port`. When connecting to a multi-DC cluster, ensure that contact points from the local datacenter are specified first in this list. | `127.0.0.1`
**cassandra_port** | The port on which your nodes are listening on. All your nodes and contact points must listen on the same port. Will be created if it doesn't exist. | `9042`
**cassandra_keyspace** | The keyspace to use in your cluster. | `kong`

> **cassandra_consistency** | Consistency setting to use when reading/ writing to the Cassandra cluster. | `ONE`

Reviewer: ```suggestion
**cassandra_consistency** | Consistency setting to use when reading from or writing to the Cassandra cluster. | `ONE`
```

**Description**

> By default the DNS resolver will use the standard configuration files

Reviewer: ```suggestion
By default, the DNS resolver will use the standard configuration files
```

`/etc/hosts` and `/etc/resolv.conf`. The settings in the latter file will be
overridden by the environment variables `LOCALDOMAIN` and `RES_OPTIONS` if they
have been set.

Kong will resolve hostnames as either `SRV` or `A` records (in that order, and
`CNAME` records will be dereferenced in the process).

> In case a name was resolved as an `SRV` record it will also override any given

Reviewer: ```suggestion
In case a name was resolved as an `SRV` record, it will also override any given
```

port number by the `port` field contents received from the DNS server.

The DNS options `SEARCH` and `NDOTS` (from the `/etc/resolv.conf` file) will be

> used to expand short names to fully qualified ones. So it will first try the

Reviewer: ```suggestion
used to expand short names to fully-qualified names. So it will first try the
```

> entire `SEARCH` list for the `SRV` type, if that fails it will try the `SEARCH`

Reviewer: ```suggestion
entire `SEARCH` list for the `SRV` type; if that fails, it will try the `SEARCH`
```

list for `A`, etc.

> For the duration of the `ttl`, the internal DNS resolver will loadbalance each

Reviewer: ```suggestion
For the duration of the `ttl`, the internal DNS resolver will load balance each
```

> request it gets over the entries in the DNS record. For `SRV` records the

Reviewer: ```suggestion
request it gets over the entries in the DNS record. For `SRV` records, the
```

same name/type will be synchronised to a
single query.

### Tuning & Behavior section

#### router_consistency

**Default:** `strict`

Defines whether this node should rebuild its router synchronously or
asynchronously (the router is rebuilt every time a Route or a Service is updated
via the Admin API or loading a declarative configuration file).

Accepted values are:

- `strict`: the router will be rebuilt synchronously, causing incoming requests
  to be delayed until the rebuild is finished.
- `eventual`: the router will be rebuilt asynchronously via a recurring
  background job running every second inside of each worker.

Note that `strict` ensures that all workers of a given node will always proxy
requests with an identical router, but that increased long tail latency can be
observed if frequent Routes and Services updates are expected.

> Using `eventual` will help preventing long tail latency issues in such cases,

Reviewer: ```suggestion
Using `eventual` will help prevent long tail latency issues in such cases,
```

but may cause workers to route requests differently for a short period of time
after Routes and Services updates.

#### router_update_frequency

**Default:** `1`

Defines how often the router changes are checked with a background job. When a

> change is detected, a new router will be built. By default we check for changes

Reviewer: ```suggestion
change is detected, a new router will be built. By default, it checks for changes
```

every second.

Raising this value will decrease the load on database servers and result in

> less jitter in proxy latency, with downside of longer converge time for router

Reviewer: ```suggestion
less jitter in proxy latency, with the downside of longer converge time for router
```

connections, when `pg_ssl_verify` or
`cassandra_ssl_verify` are enabled.

> See https://github.com/openresty/lua-nginx-module#lua_ssl_trusted_certificate

Reviewer: ```suggestion
See https://github.com/openresty/lua-nginx-module#lua_ssl_trusted_certificate.
```

This includes the certificates configured
for Kong's database connections.

> See https://github.com/openresty/lua-nginx-module#lua_ssl_verify_depth

Reviewer: ```suggestion
See https://github.com/openresty/lua-nginx-module#lua_ssl_verify_depth.
```

plugins not stored in the default search
path.

> See https://github.com/openresty/lua-nginx-module#lua_package_path

Reviewer: ```suggestion
See https://github.com/openresty/lua-nginx-module#lua_package_path.
```

Sets the Lua C module search path (LUA_CPATH).

> See https://github.com/openresty/lua-nginx-module#lua_package_cpath

Reviewer: ```suggestion
See https://github.com/openresty/lua-nginx-module#lua_package_cpath.
```

connection pool associated with every remote
server.

> See https://github.com/openresty/lua-nginx-module#lua_socket_pool_size

Reviewer: ```suggestion
See https://github.com/openresty/lua-nginx-module#lua_socket_pool_size.
```

NOTE: Only keywords "min", "max" and "passphrase" are supported.

**Example:**

`admin_gui_auth_password_complexity = { "min": "disabled,24,11,9,8" }`

### admin_gui_session_conf

**Default:** `nil`

**Description:**

Session Plugin Config (JSON)
Here you may specify the configuration for the Session plugin as used by
Kong Manager.

> For more information, see the Kong Session Plugin documentation.

Reviewer: add a link here 

```suggestion
For more information, see the Kong [Session Plugin](/hub/kong-inc/sessions/) documentation.
```

Prometheus server.

## Developer Portal

### portal

**Default:** `off`

**Description:**
Developer Portal Switch

Enable or disable the Dev Portal Interface and API

When enabled, Kong will expose the Dev Portal interface and
read-only APIs on the `portal_gui_listen` address,
and endpoints on the Admin API to manage assets.

> When enabled along with `portal_auth`:

Reviewer: ```suggestion
When enabled along with `portal_auth`,
```

```
portal_cors_origins = http://localhost:8003, https://localhost:8004

```
**NOTE:** 
In most cases, the Developer Portal is able to derive

> valid CORS origins by using portal_gui_protocol, portal_gui_host,

Reviewer: ```suggestion
valid CORS origins by using `portal_gui_protocol`, `portal_gui_host`,
```

> and if applicable, portal_gui_use_subdomains. In these cases,

Reviewer: ```suggestion
and if applicable, `portal_gui_use_subdomains`. In these cases,
```

> portal_cors_origins is not needed and can remain unset.

Reviewer: ```suggestion
`portal_cors_origins` is not needed and can remain unset.
```
